### PR TITLE
Add h2o bench groupby queries

### PR DESCRIPTION
**Pull request by andygrove**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-datafusion/issues/2879

 # Rationale for this change

I would like to make it easier to run h20 benchmarks

# What changes are included in this PR?

Add h20 benchmark

# Are there any user-facing changes?

No

### CODE REVIEW

File: benchmarks/src/bin/h2o.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    match opt {
+        Opt::GroupBy(config) => group_by(&config).await,
+    }
+}
+
+async fn group_by(config: &GroupBy) -> Result<()> {
+    let start = Instant::now();
+    let path = config.path.to_str().unwrap();
+    let ctx = SessionContext::new();
+    ctx.register_csv("x", path, CsvReadOptions::default())
```
Comment (Dandandan) [inline benchmarks/src/bin/h2o.rs:61]: The h2o abenchmark allows to read the data into memory (otherwise it would be mostly measuring the csv parser).

Comment (andygrove) [reply benchmarks/src/bin/h2o.rs:61]: Wow, the results were already looking pretty great but that would be even better, Do you have a link to where they mention this?

Comment (andygrove) [reply benchmarks/src/bin/h2o.rs:61]: nm, I see it now

```
Solutions are using in-memory data storage to achieve best timing.
```

Comment (Dandandan) [reply benchmarks/src/bin/h2o.rs:61]: Glad you found it. Some solutions also use specific (such as dictionary) encoding to get better results and lower memory usage.

The old PR for the rust solution is here:
https://github.com/h2oai/db-benchmark/pull/182

This also includes some performance tweaks like changing the allocator, compile flags, prepartioning and batch size.

**Comment by andygrove** [comment]
## Reading from CSV

```
$ ../target/release/h2o group-by --query 1 --path /mnt/bigdata/h2oai/N_1e7_K_1e2_single.csv
Running benchmarks with the following options: GroupBy(GroupBy { query: 1, path: "/mnt/bigdata/h2oai/N_1e7_K_1e2_single.csv", debug: false, mem_table: false })
Executing select id1, sum(v1) as v1 from x group by id1
h2o groupby query 1 took 1793 ms
```

## Reading from memory

```
$ ../target/release/h2o group-by --query 1 --path /mnt/bigdata/h2oai/N_1e7_K_1e2_single.csv --mem-table
Running benchmarks with the following options: GroupBy(GroupBy { query: 1, path: "/mnt/bigdata/h2oai/N_1e7_K_1e2_single.csv", debug: false, mem_table: true })
Executing select id1, sum(v1) as v1 from x group by id1
h2o groupby query 1 took 396 ms
```

**Comment by andygrove** [comment]
@Dandandan This is ready for another review

**Review by Dandandan — APPROVED** [review]

**Pull request merged**